### PR TITLE
Cleanup: remove prometheus dependencies for scheduler

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Just find that scheduler still depends prometheus directly which should be deprecated. Let us remove it.

**Which issue(s) this PR fixes**:

Part of #89267

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/runtime/framework_test.go [hunk 1]
```diff
@@ -2216,94 +2214,68 @@ func injectNormalizeRes(inj injectedResult, scores framework.NodeScoreList) *fra
...
 
 func collectAndComparePermitWaitDuration(t *testing.T, wantRes string) {
-	m := collectHistogramMetric(metrics.PermitWaitDuration)
+	m := metrics.PermitWaitDuration.WithLabelValues(wantRes)
 	if wantRes == "" {
-		if m != nil {
-			t.Errorf("PermitWaitDuration shouldn't be recorded but got %+v", m)
+		count, err := testutil.GetHistogramMetricCount(m)
+		if err != nil {
+			t.Errorf("Failed to get %s sampleCount, err: %v", metrics.PermitWaitDuration.Name, err)
```
Comment (damemi) [inline pkg/scheduler/framework/runtime/framework_test.go:2260]: This section (and the section in the `else` block that matches) can be broken out of the if statement, correct?

File: pkg/scheduler/framework/runtime/framework_test.go [hunk 2]
```diff
@@ -2216,94 +2214,68 @@ func injectNormalizeRes(inj injectedResult, scores framework.NodeScoreList) *fra
...
-	for _, p := range m.Label {
-		gotLabels[p.GetName()] = p.GetValue()
+	count, err := testutil.GetHistogramMetricCount(m)
+	if err != nil {
+		t.Errorf("Failed to get %s sampleCount, err: %v", metrics.FrameworkExtensionPointDuration.Name, err)
 	}
-	wantLabels := map[string]string{
-		"extension_point": wantExtensionPoint,
-		"status":          wantStatus.String(),
-		"profile":         testProfileName,
```
Comment (damemi) [inline pkg/scheduler/framework/runtime/framework_test.go:2256]: Do we still need to compare the wanted labels to the received labels? It looks like that was removed

Comment (gavinfish) [reply pkg/scheduler/framework/runtime/framework_test.go:2256]: `metrics.FrameworkExtensionPointDuration.WithLabelValues(wantExtensionPoint, wantStatus.String(), testProfileName)` will only fetch the values with the target labels, if the values are not matched, then either we lose the metrics or with wrong labels.

File: pkg/scheduler/framework/runtime/framework_test.go (hunk 1 above)
Comment (gavinfish) [reply pkg/scheduler/framework/runtime/framework_test.go:2260]: yes, updated.

**Comment by RainbowMango** [comment]
/approve
for component-base.

**Comment by gavinfish** [comment]
/assign @ahg-g 
for scheduler

**Comment by ahg-g** [comment]
/lgtm
/approve

Thanks!

**Comment by gavinfish** [comment]
/assign @thockin
Both component-base and scheduler teams have approved, please take a look.

**Comment by ehashman** [comment]
/cc @liggitt 

can you PTAL?

**Pull request merged**